services/worker-manager/src/data.js

```diff
@@ -154,6 +159,22 @@ class WorkerPool {
     ];
     return _.isEqual(_.pick(other, fields), _.pick(this, fields));
   }
+
+  // guess what type of worker is being used
+  // we assume that provider only contains launch configs for the workers of the same type
+  guessWorkerType() {
+    if (!Array.isArray(this.config?.launchConfigs)) {
+      return 'unknown';
+    }
+    if (this.config.launchConfigs.some(cfg =>
+      // some worker pools with docker worker might include genericWorker config by mistake with 1-2 entries
```

Comment:
this assumption was correct at the time of implementation, but then owners were notified and those worker pool configs should be fixed, so  maybe we don't need an extra check for the number of keys in that config?